build/tests/version-discovery.test.mjs

```diff
@@ -0,0 +1,150 @@
+    });
+
+    it('should respect cutoff parameter', async () => {
+      if (!coreRepoPath) {
+        console.log('Skipping test: No CORE repository path provided');
+        return;
+      }
+
+      const cutoff = 2;
+      const { versions } = await discoverVersions(coreRepoPath, cutoff);
+      const versionCount = versions.filter(v => v !== 'latest').length;
+      expect(versionCount).toBeLessThanOrEqual(cutoff);
+    });
+
+    it('should return major.minor format for retired versions', async () => {
+      if (!coreRepoPath) {
+        console.log('Skipping test: No CORE repository path provided');
+        return;
+      }
+
+      const { retired } = await discoverVersions(coreRepoPath, 2);
+      retired.forEach(version => {
+        expect(version).toMatch(/^\d+\.\d+$/);
+      });
+    });
+
+    it('should return full semver for active versions', async () => {
+      if (!coreRepoPath) {
+        console.log('Skipping test: No CORE repository path provided');
+        return;
+      }
+
+      const { versions } = await discoverVersions(coreRepoPath, 2);
+      const semverVersions = versions.filter(v => v !== 'latest');
+      semverVersions.forEach(version => {
+        expect(version).toMatch(/^v?\d+\.\d+\.\d+/);
+      });
+    });
+  });
+
+  describe('getStarlightVersions', () => {
+    it('should format versions with major.minor slugs', async () => {
+      if (!coreRepoPath) {
+        console.log('Skipping test: No CORE repository path provided');
+        return;
+      }
+
+      const starlightVersions = await getStarlightVersions(coreRepoPath, 2);
+      starlightVersions.forEach(v => {
+        expect(v).toHaveProperty('slug');
+        expect(v).toHaveProperty('label');
+        expect(v.slug).toMatch(/^v\d+\.\d+$/);
+      });
+    });
+
+    it('should not include "latest" in starlight versions', async () => {
+      if (!coreRepoPath) {
+        console.log('Skipping test: No CORE repository path provided');
+        return;
+      }
+
+      const starlightVersions = await getStarlightVersions(coreRepoPath, 2);
+      const hasLatest = starlightVersions.some(v =>
+        v.slug === 'latest' || v.label === 'latest'
+      );
+      expect(hasLatest).toBe(false);
+    });
+
+    it('should use full version for label', async () => {
+      if (!coreRepoPath) {
+        console.log('Skipping test: No CORE repository path provided');
+        return;
+      }
+
+      const starlightVersions = await getStarlightVersions(coreRepoPath, 2);
+      starlightVersions.forEach(v => {
+        expect(v.label).toMatch(/^v?\d+\.\d+\.\d+/);
+      });
+    });
+
+    it('should filter out prerelease versions', async () => {
+      if (!coreRepoPath) {
+        console.log('Skipping test: No CORE repository path provided');
+        return;
+      }
+
+      const starlightVersions = await getStarlightVersions(coreRepoPath, 2);
+      starlightVersions.forEach(v => {
+        expect(v.label).not.toMatch(/-alpha|-beta|-rc/);
```

Comment:
To my knowledge, Pepr doesn't really do pre-release (but we do release a `nightly` build). Should this just reject anything after a `x.y.z` version number instead?